### DISCUSSION TITLE
Functionality Discussion: Getting list of actively blocked links outside of the browser

### DISCUSSION THREAD

**Discussion by ayham-1** [uBlockOrigin/uBlock-issues#3412] [Ideas] [answered]
Is the project maintainer(s) willing to accept a feature that allows external user-run applications to receive what is actively being blocked by the extension?

I am willing to work on the feature myself.

**Comment by ghost** [comment]
If you only want the links, then you can do this in your own extension by watching for blocked requests (<https://github.com/gorhill/uBO-Scope>). Exporting this from uBO will require the logger to be enabled and working all the time which will probably be slow. API access into uBO internals was declined before <https://github.com/gorhill/uBlock/issues/3352>.

***

Even gathering statistics was declined before https://github.com/uBlockOrigin/uBlock-issues/issues/261#issuecomment-427926618

**Reply by gorhill** [reply, marked as answer]
> your own extension by watching for blocked requests

Right, just registering the [appropriate listeners](https://developer.chrome.com/docs/extensions/reference/api/webRequest) in your standalone extension you can find out whether a network request was allowed to reach a remote server or not -- and this works regardless of which content blocker is installed:

> The web request API guarantees that for each request, either `onCompleted` or `onErrorOccurred` is fired as the final event with one exception: If a request is redirected to a `data://` URL, `onBeforeRedirect` is the last reported event.

**Reply by ayham-1** [reply]
> API access into uBO internals was declined before [gorhill/uBlock#3352](https://github.com/gorhill/uBlock/issues/3352).
> 
> Even gathering statistics was declined before [#261 (comment)](https://github.com/uBlockOrigin/uBlock-issues/issues/261#issuecomment-427926618)

Ok that is pretty understandable.

Watching for blocked requests would be acceptable for my use case.

Thank you for your replies, I guess this discussion is solved.

**Reply by garry-ut99** [reply]
> gwarser: Even gathering statistics was declined before [#261 (comment)](https://github.com/uBlockOrigin/uBlock-issues/issues/261#issuecomment-427926618)

But it was then implemented later as experimental in 05.2019: [commit](https://github.com/manhhomienbienthuy/uBlock/commit/26708b37c1e58046f380701b364b094136ce432f) ==> [wiki](https://github.com/gorhill/uBlock/wiki/The-logger/55cdf26929a7f8f72a4f2f766efb6ae901b0fd63#filter-statistics) , 
and then it was disabled / postponed temporarily after some time in 02.2021: [commit](https://github.com/gorhill/uBlock/commit/5a5523c0b53697f816d942a1b3a712980bc77b93) ==> [wiki](https://github.com/gorhill/uBlock/wiki/The-logger/6eed5b212a6209f58cd573e10da40eb9b6e68897#filter-statistics)
